test/Sdk.Generator.Tests/FunctionMetadataProviderGeneratorTests/HttpTriggerTests.cs

```diff
@@ -0,0 +1,319 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.Functions.Worker.Sdk.Generators;
+using Xunit;
+
+namespace Microsoft.Azure.Functions.SdkGeneratorTests
+{
+    public partial class FunctionMetadataProviderGeneratorTests
+    {
+        public class HttpTriggerTests
+        {
+            private Assembly[] _referencedExtensionAssemblies;
+
+            public HttpTriggerTests()
+            {
+                // load all extensions used in tests (match extensions tested on E2E app? Or include ALL extensions?)
+                var abstractionsExtension = Assembly.LoadFrom("Microsoft.Azure.Functions.Worker.Extensions.Abstractions.dll");
+                var httpExtension = Assembly.LoadFrom("Microsoft.Azure.Functions.Worker.Extensions.Http.dll");
+                var hostingExtension = Assembly.LoadFrom("Microsoft.Extensions.Hosting.dll");
+                var diExtension = Assembly.LoadFrom("Microsoft.Extensions.DependencyInjection.dll");
+                var hostingAbExtension = Assembly.LoadFrom("Microsoft.Extensions.Hosting.Abstractions.dll");
+                var diAbExtension = Assembly.LoadFrom("Microsoft.Extensions.DependencyInjection.Abstractions.dll");
+
+                _referencedExtensionAssemblies = new[]
+                {
+                    abstractionsExtension,
+                    httpExtension,
+                    hostingExtension,
+                    hostingAbExtension,
+                    diExtension,
+                    diAbExtension
+                };
+            }
+
+            [Fact]
+            public async Task GenerateSimpleHttpTriggerMetadataTest()
+            {
+                string inputCode = @"
+            using System;
+            using System.Collections.Generic;
+            using System.Net;
+            using Microsoft.Azure.Functions.Worker;
+            using Microsoft.Azure.Functions.Worker.Http;
+
+            namespace FunctionApp
+            {
+                public static class HttpTriggerSimple
```

Comment:
Let's also add a test with nested class.

````
namespace March
{
    public class Friday
    {
        public class Foo
        {
            [Function("Function1")]
            public HttpResponseData Run(
               [HttpTrigger(AuthorizationLevel.Anonymous, "get", "post")] 
                        HttpRequestData req)
            {
            }
      }
}
````

Reply:
I tested it out (passed), but I don't actually think it's necessary - the SyntaxReceiver will find all methods with attributes on them and add to a list of candidates that the parser will go through and look for functions in.